sdks/python/apache_beam/io/hdfs_integration_test/Dockerfile

```diff
@@ -24,22 +24,13 @@ FROM $BASE_IMAGE
 
 WORKDIR /app
 ENV HDFSCLI_CONFIG /app/sdks/python/apache_beam/io/hdfs_integration_test/hdfscli.cfg
-RUN pip install --no-cache-dir holdup gsutil
-RUN gsutil cp gs://dataflow-samples/shakespeare/kinglear.txt .
 
-# Install Beam and dependencies.
-ADD sdks/python /app/sdks/python
-ADD model /app/model
-RUN cd sdks/python && \
-    python setup.py sdist && \
-    pip install --no-cache-dir $(ls dist/apache-beam-*.tar.gz | tail -n1)[gcp]
+# Add Beam SDK sources.
+COPY sdks/python /app/sdks/python
+COPY model /app/model
+
+# This step should look like setupVirtualenv minus virtualenv creation.
+RUN pip install --no-cache-dir tox==3.11.1 -r sdks/python/build-requirements.txt
```

Comment:
so our goal is to install `grpcio-tools` (and therefore protobuf, one of its deps) into the same python environment where sdist is run, _before the tests run_.  This is easier said than done.  

Installing them here in the dockerfile (via build-requirements.txt) affects the environment where tox is installed and runs, but i'm not sure if it affects the environments where tox runs sdist, especially if tox is trying to create a temp env to run sdist (which I _know_ it does when it's operating in pep517 mode).  

Adding build-requirements.txt to the deps in tox.ini I think only affects the environment where the tests run, but not where tox runs sdist.

gen_protos.py also does its own attempt to install build-requirements.txt if it detects `grpcio-tools` is not installed, but it does it in a weird way with a target install directory, which I think might not be reliable for the google .pth file that makes protobufs importable in python2 (see earlier discussion about site dirs).

So, I've got 2 ideas left for how to hack this together without going full pep517:

1) run `python setup.py sdist` here in dockerland, after installing build-requirements, and pass the sdist to tox.  That should guarantee that sdist is run in an env where protobuf is properly installed where the google .pth file will work, since we're installing it into the site-packages of the python interpreter running here, and not some temp dir. 
2) change the logic in gen_protos where it installs build-requirements to also ensure that the directory it installs into is added as a site dir, by calling `site.addsitedir()` on that directory from the environment where we want to be able to import `google.protobuf`.

Reply:
Ah, I wrote my comment on the previous version of this.  Glad to see that adding the build-requirements here worked.  Like I said I wasn't sure if that would affect how tox runs sdist, since when I was testing with pyproject.toml, it did not, because it was using an isolated build environment.